### DISCUSSION TITLE
[5.0]: Improve editing functions (UX) of matrix fields

### DISCUSSION THREAD

**Discussion by alexanderbuergin** [craftcms/cms#14178] [Ideas] [answered]
**Introduction of Copy, Paste, and Duplicate Functions for Matrix Fields**

I propose integrating the ability to duplicate, copy, and paste blocks within matrix fields. This feature would significantly streamline content management, allowing for efficient replication and modification of existing blocks.

<img width="1431" alt="Bildschirmfoto 2024-01-23 um 21 32 53" src="https://github.com/craftcms/cms/assets/85960512/b009292a-dd28-4b1a-a2b4-b46e892bc0de">

**Cross-Entry Block Copying Capability**

It would be highly beneficial to enable the copying of blocks across different entries. This functionality would aid in maintaining consistency and efficiency when working with multiple entries that share similar content structures.

**Global Entry Type Compatibility for Matrix Fields**

Given that matrix fields are now linked to global entry types, it would be advantageous to allow the pasting of blocks into different matrix fields, provided they utilize the same entry types.

**Comment by scoch5** [comment]
Personally I think this feature would be really useful if not essential.
In addition to pasting inside, perhaps you could also enable dragging inside another block, a bit like Neo does.

**Comment by scoch5** [comment]
@brandonkelly  I'm doing some data entry with the new matrix and I realize more and more how indispensable this functionality is to improve the authors experience.

**Comment by alexanderloewe** [comment]
+1

**Comment by olsp** [comment]
Do you know how much my clients would love me when duplicating blocks would be possible?

**Reply by alexanderloewe** [reply]
We need this.

**Comment by turnstylerj** [comment]
This is a killer feature of Neo, would love to see it available for new Matrix.

**Comment by garycrane** [comment]
+1 for this feature. Would be interested to know if this is on the roadmap? Every client has asked me for it at one point or another. If Matrix is being used as a page builder, it would make the authoring experience so much better.

**Comment by alexanderbuergin** [comment]
@brandonkelly: Any Update on that one?

**Comment by davidhellmann** [comment]
There is a plugin that handles a few of the requested features: https://plugins.craftcms.com/matrix-extended?craft5

**Reply by alexanderloewe** [reply]
I know, but i think this should be a core option/function

**Reply by garycrane** [reply]
Also, it doesn't appear to work when the Matrix field is configured with card view. Only works when inline-editable. I found Matrix Extended and Smith by Verbb to be hit & miss. Both have weird quirks/bugs that make them difficult to include on a production site. Shame.

**Comment by brandonkelly** [comment, marked as answer]
Nested element duplication is coming in Craft 5.7 :tada: (#16819)

~~Agree that copy/pasting would be good to have as well, so going to leave this discussion open for those.~~

**Update:** copy/paste support has been added for 5.7 as well! (#16888)

**Reply by ryanleichty** [reply]
Awesome! Definitely a fan of having copy/paste too—that's something our author’s are currently using with Neo.

**Reply by angelokeirsebilck** [reply]
Agree, duplicating is helpful. Ability to copy/paste blocks from one matrix to another matrix field on another entry would be a game changer.

**Reply by garycrane** [reply]
+1 for Copy/Paste too. I'm using it lots via Matrix Extended but native would be amazing.

**Reply by alexanderbuergin** [reply]
Yes please

**Reply by brandonkelly** [reply]
Copy/paste support has been added for 5.7 as well! (#16888)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with nested element duplication and copy-paste support 🎉
